File: WEEK 1/DAY 1 - Array Basics/SOLUTIONS/02_Medium_Problem_Solution/Solution.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var maxSubArray = function(nums) {
    let max_sum = nums[0];
    let current_sum = 0;
    
    for (let num of nums) {
        current_sum = Math.max(num, current_sum + num);
        max_sum = Math.max(max_sum, current_sum);
    }
    
    return max_sum;
};

File: WEEK 1/DAY 1 - Array Basics/SOLUTIONS/02_Medium_Problem_Solution/Solution.cpp
#include <vector>
#include <algorithm>

class Solution {
public:
    int maxSubArray(std::vector<int>& nums) {
        int max_sum = nums[0];
        int current_sum = 0;

        for (int num : nums) {
            current_sum = std::max(num, current_sum + num);
            max_sum = std::max(max_sum, current_sum);
        }

        return max_sum;
    }
};
